### ISSUE TITLE
TypeError: Cannot read properties of null (reading 'parentNode') when using keepAlive

### ISSUE THREAD

**Issue by vloss3** [nuxt/nuxt#25751]
### Environment

Working directory: /home/projects/nuxt-starter-f4vru5
Nuxt project info:
------------------------------
- Operating System: Linux
- Node Version:     v18.18.0
- Nuxt Version:     3.10.1
- CLI Version:      3.10.0
- Nitro Version:    2.8.1
- Package Manager:  npm@10.2.3
- Builder:          -
- User Config:      devtools
- Runtime Modules:  -
- Build Modules:    -
------------------------------

### Reproduction

https://stackblitz.com/edit/nuxt-starter-f4vru5?file=pages%2Findex.vue

### Describe the bug

The error `TypeError: Cannot read properties of null (reading 'parentNode')` appears in the console when navigating between pages if `<NuxtPage :keepalive="{ max: 5 }" />` and NuxtLayout are used.
Steps:

1. Open dev tools console
2. Navigate between pages, e.g. Index -> Node 1 -> Index
3. Check the console for errors

### Logs

```shell-script
chunk-JSACULTC.js?v=017ed9e0:9626 Uncaught (in promise) TypeError: Cannot read properties of null (reading 'parentNode')
    at parentNode (chunk-JSACULTC.js?v=017ed9e0:9626:30)
    at ReactiveEffect.componentUpdateFn [as fn] (chunk-JSACULTC.js?v=017ed9e0:7537:11)
    at ReactiveEffect.run (chunk-JSACULTC.js?v=017ed9e0:430:19)
    at instance.update (chunk-JSACULTC.js?v=017ed9e0:7577:17)
    at callWithErrorHandling (chunk-JSACULTC.js?v=017ed9e0:1651:32)
    at flushJobs (chunk-JSACULTC.js?v=017ed9e0:1858:9)
```

**Comment by danielroe** [comment]
This is likely an upstream Vue issue.

**Comment by vloss3** [comment]
Related Vue upstream:
Issue: https://github.com/vuejs/core/issues/10098
PR: https://github.com/vuejs/core/pull/10184

**Comment by huang-julien** [comment]
I think a workaround is to set your vue version to 3.4.8 🤔

**Comment by vloss3** [comment]
Fixed in Vue [v3.4.20](https://github.com/vuejs/core/releases/tag/v3.4.20), refresh package-lock to use the latest Vue version.
